sky/engine/core/dart/text.dart

```diff
@@ -0,0 +1,216 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart_ui;
+
+/// Whether to slant the glyphs in the font
+enum FontStyle {
+  /// Use the upright glyphs
+  normal,
+
+  /// Use glyphs designed for slanting
+  italic,
+}
+
+/// The thickness of the glyphs used to draw the text
+enum FontWeight {
+  /// Thin, the least thick
+  w100,
+
+  /// Extra-light
+  w200,
+
+  /// Light
+  w300,
+
+  /// Normal / regular / plain
+  w400,
+
+  /// Medium
+  w500,
+
+  /// Semi-bold
+  w600,
+
+  /// Bold
+  w700,
+
+  /// Extra-bold
+  w800,
+
+  /// Black, the most thick
+  w900
+}
+
+/// Whether to align text horizontally
+enum TextAlign {
+  /// Align the text on the left edge of the container
+  left,
+
+  /// Align the text on the right edge of the container
+  right,
+
+  /// Align the text in the center of the container
+  center
+}
+
+/// A horizontal line used for aligning text
+enum TextBaseline {
+  // The horizontal line used to align the bottom of glyphs for alphabetic characters
+  alphabetic,
+
+  // The horizontal line used to align ideographic characters
+  ideographic
+}
+
+/// A linear decoration to draw near the text
+enum TextDecoration {
+  /// Do not draw a decoration
+  none,
+
+  /// Draw a line underneath each line of text
+  underline,
+
+  /// Draw a line above each line of text
+  overline,
+
+  /// Draw a line through each line of text
+  lineThrough
+}
+
+/// The style in which to draw a text decoration
+enum TextDecorationStyle {
+  /// Draw a solid line
+  solid,
+
+  /// Draw two lines
+  double,
+
+  /// Draw a dotted line
+  dotted,
+
+  /// Draw a dashed line
+  dashed,
+
+  /// Draw a sinusoidal line
+  wavy
+}
+
+// This encoding must match ParagraphBuilder::pushStyle.
```

Comment:
I assume you don't mean the ParagraphBuilder::pushStyle in this file, but the C++ version? Clarify the comment.